src/firmware_import_export.py

```diff
@@ -0,0 +1,195 @@
+        included_files = db.get_all_files_in_fw(fw_uid)
+        with BytesIO() as buffer:
+            with ZipFile(buffer, 'w', ZIP_DEFLATED) as zip_file:
+                file_task = progress.add_task('Fetching files', total=len(included_files) + 1)
+                for fo_uid in included_files.union({fw_uid}):
+                    file_path = self.fs_organizer.generate_path_from_uid(fo_uid)
+                    zip_file.writestr(f'files/{fo_uid}', Path(file_path).read_bytes())
+                    progress.advance(file_task)
+                progress.remove_task(file_task)
+                zip_file.writestr(
+                    'data.json',
+                    json.dumps(self._fetch_db_data(fw_uid, included_files, db, progress)),
+                )
+            target_path = self.target_dir / f'FACT_export_{fw_uid}.zip'
+            target_path.write_bytes(buffer.getvalue())
+            logger.info(f'Exported firmware {fw_uid} to {target_path}')
+
+    @staticmethod
+    def _fetch_db_data(uid: str, all_files: set[str], db, progress: Progress) -> dict:
+        db_data = {
+            'db_revision': get_current_revision(),
+            'files': [],
+            'firmware': db.get_object(uid).to_json(),
+            'uid': uid,
+        }
+        db_task = progress.add_task('Fetching DB entries', total=len(all_files))
+        for fo in db.get_objects_by_uid_list(all_files):
+            db_data['files'].append(fo.to_json(vfp_parent_filter=all_files.union({uid})))
+            progress.advance(db_task)
+        progress.remove_task(db_task)
+        return db_data
+
+
+class FwImporter:
+    def __init__(self, force: bool):
+        self.db_interface = BackendDbInterface()
+        self.fs_organizer = FSOrganizer()
+        self.force = force
+        self.progress: Progress | None = None
+
+    def import_files(self, file_list: list[str]):
+        with Progress(*COLUMNS) as progress:
+            self.progress = progress
+            import_task = progress.add_task('Importing files', total=len(file_list))
+            for file in file_list:
+                path = Path(file)
+                if not path.is_file():
+                    logging.error(f'File {path} does not exist')
+                self._import_file(path)
+                progress.advance(import_task)
+        self.progress = None
+
+    def _import_file(self, path: Path):
+        with ZipFile(path, 'r') as zip_file:
+            if 'data.json' not in zip_file.namelist():
+                logging.error('Error: data.json not found in uploaded import file')
+                return
+            try:
+                data = json.loads(zip_file.read('data.json'))
+            except json.JSONDecodeError as error:
+                logging.error(f'Error: data.json is not a valid JSON file: {error}')
+                return
+            if not all(k in data for k in ['db_revision', 'files', 'firmware', 'uid']):
```

Comment:
Perhaps we should explain in the description that a data.json is mandatory and which keys it must contain. Otherwise, users who want to use the tool to import firmware (which has not been exported before) will have to look at the code to find out which keys are needed